Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1539 @@
+#include <algorithm>
+#include <boost/filesystem.hpp>
+#include <boost/make_shared.hpp>
+#include <cxxtest/TestSuite.h>
+#include <fstream>
+#include <gmock/gmock.h>
+#include <iostream>
+#include <memory>
+#include <set>
+
+#define DEBUG_PRINT(x) std::cout << "\nDEBUG: " << x << std::endl
+
+using namespace Mantid::NexusGeometry;
+
+//---------------------------------------------------------------
+namespace {
+
+typedef std::vector<std::string> fullH5Path;
+
+// get path as string. Used for the dependency tests.
+std::string toH5PathString(fullH5Path &path) {
+  std::string pathString = "";
+  for (const std::string &grp : path) {
+    pathString += "/" + grp;
+  }
+  return pathString;
+}
+
+const H5G_obj_t GROUP_TYPE = static_cast<H5G_obj_t>(0);
+const H5G_obj_t DATASET_TYPE = static_cast<H5G_obj_t>(1);
+
+const std::string SHORT_NAME = "short_name";
+const std::string NX_CLASS = "NX_class";
+const std::string NX_ENTRY = "NXentry";
+const std::string NX_INSTRUMENT = "NXinstrument";
+const std::string NX_SOURCE = "NXsource";
+const std::string NX_SAMPLE = "NXsample";
+const std::string SHAPE = "shape";
+
+const std::string NX_TRANSFORMATIONS = "NXtransformations";
+const std::string NX_CHAR = "NX_CHAR";
+
+const std::string TRANSFORMATION_TYPE = "transformation_type";
+const std::string ROTATION = "rotation";
+const std::string TRANSLATION = "translation";
+const std::string TRANSFORMATIONS = "transformations";
+const std::string VECTOR = "vector";
+const std::string LOCATION = "location";
+const std::string DEPENDS_ON = "depends_on";
+const std::string ORIENTATION = "orientation";
+const std::string UNITS = "units";
+const std::string METRES = "m";
+const std::string NAME = "name";
+const std::string X_PIXEL_OFFSET = "x_pixel_offset";
+const std::string Y_PIXEL_OFFSET = "y_pixel_offset";
+const std::string Z_PIXEL_OFFSET = "z_pixel_offset";
+const std::string SELF_DEPENDENT = ".";
+
+const std::string DEFAULT_ROOT_PATH = "raw_data_1";
+
+class MockProgressBase : public Mantid::Kernel::ProgressBase {
+public:
+  GNU_DIAG_OFF_SUGGEST_OVERRIDE
+  MOCK_METHOD1(doReport, void(const std::string &));
+  GNU_DIAG_OFF_SUGGEST_OVERRIDE
+};
+
+//  local Class used for validation of the structure of a nexus file as needed
+//  for the unit tests.
+class HDF5FileTestUtility {
+
+public:
+  HDF5FileTestUtility(const std::string &fullPath) {
+    boost::filesystem::path tmp = fullPath;
+
+    if (!boost::filesystem::exists(tmp)) {
+      throw std::invalid_argument("no such file.\n");
+    } else {
+      m_file.openFile(fullPath, H5F_ACC_RDONLY);
+    }
+  }
+
+  /* safely open a HDF5 group path with additional helpful
+   debug information to output where open fails) */
+  H5::Group openfullH5Path(const fullH5Path &pathList) const {
+
+    H5::Group child;
+    H5::Group parent;
+    try {
+      parent = m_file.openGroup(pathList[0]);
+    } catch (H5::Exception &) {
+      DEBUG_PRINT("HDF5 Exception was thrown at open root group.");
```

Comment:
Not sure about the catches and DEBUG prints here. I would just let things fail personally.